src/Umbraco.Abstractions/Security/IWebSecurity.cs

```diff
@@ -0,0 +1,92 @@
+using Umbraco.Core;
+using Umbraco.Core.Models.Membership;
+
+namespace Umbraco.Web.Security
+{
+    public interface IWebSecurity
+    {
+        /// <summary>
+        /// Gets the current user.
+        /// </summary>
+        /// <value>The current user.</value>
+        IUser CurrentUser { get; }
+
+        /// <summary>
+        /// Logs a user in.
+        /// </summary>
+        /// <param name="userId">The user Id</param>
+        /// <returns>returns the number of seconds until their session times out</returns>
+        double PerformLogin(int userId);
```

Comment:
We need to remove this, it's not actually used for login, i'll mark as obsolete for now